### DISCUSSION TITLE
New update not support cp-311

### DISCUSSION THREAD

**Discussion by JasonHe-WQ** [openai/whisper#1043] [Q&A] [answered]
It seems that an old module named ```numba``` does not support cp-311.<br>
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\18075\AppData\Local\Programs\Python\Python311\Scripts\whisper.exe\__main__.py", line 4, in <module>
  File "C:\Users\18075\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\__init__.py", line 13, in <module>
    from .model import ModelDimensions, Whisper
  File "C:\Users\18075\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\model.py", line 13, in <module>
    from .transcribe import transcribe as transcribe_function
  File "C:\Users\18075\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\transcribe.py", line 19, in <module>
    from .timing import add_word_timestamps
  File "C:\Users\18075\AppData\Local\Programs\Python\Python311\Lib\site-packages\whisper\timing.py", line 6, in <module>
    import numba
ModuleNotFoundError: No module named 'numba'
```
Even I tried ```pip install .\numba-0.56.4-cp310-cp310-win_amd64.whl --force-reinstall```, I was denied again.<br>

```
PS C:\Users\18075\Downloads> pip install numba --force-reinstall
Collecting numba
  Using cached numba-0.56.4.tar.gz (2.4 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\18075\AppData\Local\Temp\pip-install-ylt0ejg6\numba_da2b9edea0114b4ba568b7cc9170abcd\setup.py", line 51, in <module>
          _guard_py_ver()
        File "C:\Users\18075\AppData\Local\Temp\pip-install-ylt0ejg6\numba_da2b9edea0114b4ba568b7cc9170abcd\setup.py", line 48, in _guard_py_ver
          raise RuntimeError(msg.format(cur_py, min_py, max_py))
      RuntimeError: Cannot install on Python version 3.11.2; only versions >=3.7,<3.11 are supported.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```
So plz help me

**Comment by JasonHe-WQ** [comment]
I am sorry, ```numba``` is not old, and you can see it [here](https://github.com/numba/numba)

**Comment by jongwook** [comment]
[PyTorch only supports 3.7-3.9 on Windows](https://pytorch.org/get-started/locally/#windows-python), so you'll need to downgrade to Python 3.9. [Anaconda](https://www.anaconda.com/products/distribution#Downloads) is an easy way to install Python 3.9 on Windows.

**Reply by JasonHe-WQ** [reply]
No,  **[Pytorch Supports cp-311](https://download.pytorch.org/whl/torch)** and the Pytorch 2.0 with cp-311 and CUDA 12 is significantly faster than cp-310 or any previous.<br>
## If you can't answer clearly and rightly, plz don't answer for it may not help and make it more complex. Imagine your classmates who may not expert in conda may delete your critical env and make it horrible after reading your advice.<br> _**AND IT NEARLY HAPPENED**_<br>
## Also, you didn't give me an answer which matches my question.<br>
### I don't need you to point at my env and, moreover, conda is awful, take venv(s) instead may be a better choice<br>
I found out the numba MAY supports cp-311 [in another week](https://github.com/numba/numba/issues/8304#issuecomment-1456646575).<br>

**Reply by JasonHe-WQ** [reply]
I am sorry for what I said since I do realize numba is installed in conda as a default.

**Comment by JasonHe-WQ** [comment, marked as answer]
Just wait, they MAY [update numba in one week](https://github.com/numba/numba/issues/8304#issuecomment-1456646575).

**Reply by egeres** [reply]
Current numba progress on the next release can be followed in https://github.com/numba/numba/issues/8841!

**Reply by JasonHe-WQ** [reply]
Numba has [updated the new version ](https://github.com/numba/numba/milestone/63) and we can try it now.

**Comment by gustawdaniel** [comment]
I have this error on Linux

```
  Using cached numba-0.56.4.tar.gz (2.4 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-yogampb1/numba_c68cd26e77a64650805010d424bf5c97/setup.py", line 51, in <module>
          _guard_py_ver()
        File "/tmp/pip-install-yogampb1/numba_c68cd26e77a64650805010d424bf5c97/setup.py", line 48, in _guard_py_ver
          raise RuntimeError(msg.format(cur_py, min_py, max_py))
      RuntimeError: Cannot install on Python version 3.11.2; only versions >=3.7,<3.11 are supported.
      [end of output]
```

tried for python 3.11, 3.10 and 3.9

**Reply by JasonHe-WQ** [reply]
Sorry😣, I didn't try it on linux. But from what you pasted, it seems you are using py 3.11.2 not 3.10 or 3.9, which is not supported currently since the [numba](url) have not released new version yet, you can find it [here](https://github.com/numba/numba/issues/8304#issuecomment-1461186773).<br>
Maybe you can wait for another week. As a by stander of both projects, I can't give you any acurate support, sorry again.😢<br>

**Reply by gustawdaniel** [reply]
quite weird. I will wait for `numba` update.

![Zrzut ekranu z 2023-03-18 09-30-52](https://user-images.githubusercontent.com/16663028/226073477-c57d3bd0-cacd-48a7-ab25-3c0db343aa93.png)

**Reply by JasonHe-WQ** [reply]
You can try it [now](https://github.com/numba/numba/releases/tag/0.57.0)

**Comment by mashamzahid** [comment]
try "pip install openai-whisper" it worked for me on python 3.11 venv
